build_projects/shared-build-targets-utils/Utils/Crossgen.cs

```diff
@@ -198,6 +198,9 @@ public void CrossgenDirectory(string sharedFxPath, string pathToAssemblies)
                 { "COMPlus_PartialNGen", "0" }
             };
 
+            // Form the dynamic path that would not collide if another instance of this is running.
+            string basePath = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
```

Comment:
+1 to @schellap's comment.  It should be simple enough as just not using Path.GetTempPath() and using obj folder, right?  You can still use the guid if there were races?